file_scanning/clamd/clamd.go

```diff
@@ -0,0 +1,229 @@
+package clamd
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"net/url"
+	"strings"
+)
+
+const (
+	RES_OK          = "OK"
+	RES_FOUND       = "FOUND"
+	RES_ERROR       = "ERROR"
+	RES_PARSE_ERROR = "PARSE ERROR"
+)
+
+type Clamd struct {
+	address string
+}
+
+type Stats struct {
+	Pools    string
+	State    string
+	Threads  string
+	Memstats string
+	Queue    string
+}
+
+type ScanResult struct {
+	Raw         string
+	Description string
+	Path        string
+	Hash        string
+	Size        int
+	Status      string
+}
+
+var EICAR = []byte(`X5O!P%@AP[4\PZX54(P^)7CC)7}$EICAR-STANDARD-ANTIVIRUS-TEST-FILE!$H+H*`)
+
+func (c *Clamd) newConnection() (conn *CLAMDConn, err error) {
+
+	var u *url.URL
+
+	if u, err = url.Parse(c.address); err != nil {
+		return
+	}
+
+	conn, err = newCLAMDTcpConn(u.Host)
+
+	return
+}
+
+func (c *Clamd) simpleCommand(command string) (chan *ScanResult, error) {
+	conn, err := c.newConnection()
+	if err != nil {
+		return nil, err
+	}
+
+	err = conn.sendCommand(command)
+	if err != nil {
+		return nil, err
+	}
+
+	ch, wg, err := conn.readResponse()
+
+	go func() {
+		wg.Wait()
+		conn.Close()
+	}()
+
+	return ch, err
+}
+
+/*
+Check the daemon's state (should reply with PONG).
+*/
+func (c *Clamd) Ping() error {
```

Comment:
Comment should start with the function name first 
https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences

Also based on https://go.dev/doc/effective_go#commentary while Go does support `/* */` style comments `//` is the preferred one when commenting code and `/* */` is normally reserved for package level comments

`(can be ignored)`